research/delf/delf/python/training/global_features/train.py

```diff
@@ -0,0 +1,719 @@
+# Lint as: python3
+# Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Training script for Global Features model."""
+
+import math
+import os
+import time
+
+import numpy as np
+import pickle
+import tensorflow as tf
+import tensorflow_addons as tfa
+
+from absl import flags
+from absl import app
+from absl import logging
+
+from delf.python.datasets import tuples_dataset
+from delf.python.datasets.revisited_op import dataset as testdataset
+from delf.python.training.losses import ranking_losses
+from delf.python.training import global_features_utils
+from delf.python.training import tensorboard_utils
+from delf.python.training.model import global_model
```

Comment:
I noticed the global_model part is not already checked in. Can you point to me which PR it's in, so I can review it? Sorry if I missed here :)

Reply:
same for sfm120k :)

Reply:
@andrefaraujo, here it is:
a) Dataset related modules: https://github.com/tensorflow/models/pull/9903
b) Global model definition: https://github.com/tensorflow/models/pull/9902
c) Global model training: https://github.com/tensorflow/models/pull/9904

Reply:
Thanks! Will look at them in this order.